cmd/scale.go

```diff
@@ -219,12 +220,22 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 			return fmt.Errorf("The provided resource group does not contain any vms")
 		}
 		for _, vm := range *vms.Value {
-
-			poolName, nameSuffix, index, err := utils.K8sLinuxVMNameParts(*vm.Name)
-			if err != nil || !strings.EqualFold(poolName, sc.agentPoolToScale) || !strings.EqualFold(nameSuffix, sc.nameSuffix) {
+			vmTags := *vm.Tags
+			poolName := *vmTags["poolName"]
+			nameSuffix := *vmTags["resourceNameSuffix"]
+			
+			//Changed to string contains for the nameSuffix as the Windows Agent Pools use only a substring of the first 5 characters of the entire nameSuffix
+			if err != nil || !strings.EqualFold(poolName, sc.agentPoolToScale) || !strings.Contains(sc.nameSuffix, nameSuffix) {
 				continue
 			}
 
+			osPublisher := *vm.StorageProfile.ImageReference.Publisher
```

Comment:
I think it'd be cleaner to use `*vm.StorageProfile.OsDisk.OsType == Windows` instead

Reply:
It should be cleaner, however I decided to use the osPublisher as it is the same check done for VMSS (which are not defining OsDisk.OsType).
Do you want me to change this?

Reply:
Ah that makes sense. All good then.